tests/models/test_utils.py

```diff
@@ -74,3 +74,40 @@ class CorrectModelWithOptimadeField(BaseModel):
     strict_schema = CorrectModelWithStrictField.schema()
     strict_schema["title"] = optimade_schema["title"]
     assert strict_schema == optimade_schema
+
+
+def test_formula_regexp():
+    """This test checks some simple chemical formulae with the
+    `CHEMICAL_FORMULA_REGEXP`.
+
+    """
+    import re
+    from optimade.models.utils import CHEMICAL_FORMULA_REGEXP
+
+    class DummyModel(BaseModel):
+        formula: str = Field(regex=CHEMICAL_FORMULA_REGEXP)
+
+    good_formulae = (
+        "AgCl",
+        "H5F",
+        "LiP5",
+        "Jn7Qb4",  # Regexp does not care about the actual existence of elements
```

Comment:
True, but the validation of `chemical_formula_reduced` and `chemical_formula_hill` does care as mentioned above. I know this is not what is being tested here, but the simple existence of this item in `good_formulae` tells me you might feel the same as me concerning my comment above?

Reply:
I disagree, this is listed as good formula just to make sure that the regexp does allow any elements, so that the exact allowed elements can be dealt with on a field-by-field basis (e.g. for `chemical_formula_anonymous` the allowed "elements" are A, B, C... etc).

Reply:
Fair enough.